packages/cards-editor-middleware/src/index.ts

```diff
@@ -0,0 +1,130 @@
+import type { RequestHandler, Request, Response } from 'express';
+import type { MiddlewareParameters } from '@ui5/server';
+import { json, Router as createRouter } from 'express';
+import path, { join } from 'path';
+import { existsSync, writeFileSync, mkdirSync, readFileSync } from 'fs';
+import { getWebappPath } from '@sap-ux/project-access';
+import { render } from 'ejs';
+import * as utils from './utilities';
+import os from 'os';
+
+export const enum ApiRoutes {
+    previewGeneratorSandbox = '/test/flpGeneratorSandbox.html',
+    cardsStore = '/cards/store',
+    i18nStore = '/editor/i18n'
+}
+
+module.exports = async ({ resources }: MiddlewareParameters<any>): Promise<RequestHandler> => {
+    const router = createRouter();
+    router.use(json());
+
+    const manifest = await resources.rootProject.byPath('/manifest.json');
+    if (!manifest) {
+        throw new Error('No manifest.json found.');
+    }
+
+    /**
+     * Route to serve card generator flp sandbox
+     */
+    router.get(ApiRoutes.previewGeneratorSandbox, async (_req, res: Response) => {
+        const app = JSON.parse(await manifest.getString())['sap.app'];
+        res.status(200).send(
+            render(readFileSync(join(__dirname, '../templates/editor.html'), 'utf-8'), {
+                templateModel: {
+                    appTitle: app.title || 'Card Editor Preview',
+                    component: app.id
+                }
+            })
+        );
+    });
+
+    /**
+     * Route to store card manifest files, the files are stored in the webapp folder of the project
+     * and the application manifest.json file is updated with the new card manifests information within the sap.cards.ap.embeds
+     */
+    router.post(ApiRoutes.cardsStore, async (req: Request, res: Response) => {
+        try {
+            const { floorplan, localPath, fileName = 'manifest.json', manifests } = req.body;
+            const webappPath = await getWebappPath(path.resolve());
+            const fullPath = path.resolve(webappPath, localPath);
+
+            if (!fullPath.startsWith(webappPath)) {
+                throw new Error('Invalid path');
+            }
+
+            const file = utils.prepareFileName(fullPath + '/' + fileName);
+            if (!existsSync(fullPath)) {
+                try {
+                    mkdirSync(fullPath, { recursive: true });
+                } catch (err) {
+                    res.status(403).send(`Files could not be created/updated.`);
+                }
+            }
+            const multipleCards = utils.prepareCardTypesForSaving(manifests);
+            writeFileSync(join(fullPath, file), multipleCards.integration);
+            writeFileSync(join(fullPath, 'adaptive-' + file), multipleCards.adaptive);
+
+            const manifestPath = './webapp/manifest.json';
```

Comment:
Please do not hard-code folder `webapp`. Above we correctly use `await getWebappPath()`.

Reply:
fixed now.